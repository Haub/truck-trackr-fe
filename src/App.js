import React, { Component } from 'react';

class App extends Component {
  render() {
    return (
      <div>
        <header>
        </header>
      </div>
    );
  }
}

export default App;
