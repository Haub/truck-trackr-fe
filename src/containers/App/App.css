.main {
  background-color: #cddddc;
  display: grid;
  grid-template-rows: 110px 1fr;
  height: 100vh;
}
header {
  background-color: #feecc8;
  padding: 1rem;
  grid-row: 1 / span 1;
}

main {
  grid-row: 2 / span 1;
}

#wrapper {
    width: 240px;
    padding-left: 0.5rem;
    height: 100vh;
    position: absolute;
    left:0px;
    top: 0px;
    transition: all 0.7s;
}

.main-title {
  font-family: "Averia Libre", cursive;
  text-align: center;
  color: #6d3e48;
  font-size: 4.5rem;
  letter-spacing: 0.2em;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
}
section {
  position: relative;
  bottom: 70px;
  height: 200px;
  box-sizing: border-box;
}

section:after {
  position: absolute;
  top: 1.5em;
  left: 1.5em;
  width: 40px;
  height: 40px;
  border: 1px solid #6D3E48;
  color: #6D3E48;
  font-size: 20px;
  text-align: center;
  line-height: 20px;
}
.hamburger-trigger,
.hamburger-trigger span {
  display: inline-block;
  transition: all .2s;
  box-sizing: border-box;
  z-index: 5;
}
.hamburger-trigger {
  position: relative;
  width: 50px;
  height: 44px;
}
.hamburger-trigger.true span {
  border: 2px solid #FEECC8;
  background-color: #FEECC8;
  transition: all .2s;
}
.hamburger-trigger span {
  position: absolute;
  left: 0;
  width: 100%;
  height: 5px;
  background-color: #6D3E48;
  border-radius: 6px;
}
.hamburger-trigger span:nth-of-type(1) {
  top: 0;
}
.hamburger-trigger span:nth-of-type(2) {
  top: 20px;
}
.hamburger-trigger span:nth-of-type(3) {
  bottom: 0;
}
#hamburger.true span:nth-of-type(1) {
  -webkit-transform: translateY(20px) rotate(-45deg);
  transform: translateY(20px) rotate(-45deg);
}
#hamburger.true span:nth-of-type(2) {
  left: 50%;
  opacity: 0;
  -webkit-animation: active-hamburger-bar02 .8s forwards;
  animation: active-hamburger-bar02 .8s forwards;
}
#hamburger.true span:nth-of-type(3) {
  -webkit-transform: translateY(-20px) rotate(45deg);
  transform: translateY(-20px) rotate(45deg);
}

.truck-icon,
.barrel-icon {
  height: 65px;
  padding-left: 10px;
  padding-right: 30px;
}
.register {
  font-size: 1.2rem;
}
.content-holder {
  display: flex;
  justify-content: center;
}

.logout-button {
  position: absolute;
  right: 40px;
  top: 30px;
  width: 6.5rem;
  background-color: #6D3E48;
  border: 2px solid #8a646c;
  border-radius: 10px;
  color: #fff;
  height: 40px;
  font-size: 20px;
  cursor: pointer;
}