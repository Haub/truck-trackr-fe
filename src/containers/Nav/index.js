import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './nav.css'

export const Nav = (props) => {

    return (
        <nav>
            
        </nav>
    )
}