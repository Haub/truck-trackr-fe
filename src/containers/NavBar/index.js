import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { loadFoodTrucks, loadBreweries, loadProfile } from '../../actions';
import { mockFoodTrucks, mockBreweries } from '../../tests/testMocks';

import './NavBar.css'

export class NavBar extends Component {
    constructor() {
        super()
        this.state = {

        }
    }

    loadData = async (event) => {
        const businessType = event.target.getAttribute('name')

        try {
            const response = await fetch(`https://truck-trackr-api.herokuapp.com/api/v1/${businessType}`);
            const result = await response.json();
            if(businessType === 'breweries'){
                this.props.loadProfile({page: 'breweries'})
                this.props.loadBreweries(result.data)
            } else {
                this.props.loadProfile({page: 'food_trucks'})
                this.props.loadFoodTrucks(result.data)
            }
        } catch(error) {
            throw new Error(error.message)
        }
    }

    loadMockData = (event) => {
        const businessType = event.target.getAttribute('name')

        if(businessType === 'breweries') {
            this.props.loadBreweries(mockBreweries)
        } else if(businessType === 'food_trucks') {
            this.props.loadFoodTrucks(mockFoodTrucks)
        }
    }

    render(){
        return (
            <nav>
                <ul className='nav-controls'>
                    <li className='nav-control my-profile'>My Profile</li>
                    <NavLink to='/breweries' onClick={this.loadData} name='breweries' className='nav-control breweries'>Breweries</NavLink>
                    <NavLink to='/food_trucks' onClick={this.loadData} name='food_trucks' className='nav-control food-trucks'>Food Trucks</NavLink>
                    <input className='search-input' placeholder='Search'/>
                </ul>
            </nav>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    loadBreweries: (breweries) => dispatch(loadBreweries(breweries)),
    loadFoodTrucks: (foodTrucks) => dispatch(loadFoodTrucks(foodTrucks)) ,
    loadProfile: (profile) => dispatch(loadProfile(profile))
});

export default connect(null, mapDispatchToProps)(NavBar);