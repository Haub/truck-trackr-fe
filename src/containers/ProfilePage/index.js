import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './ProfilePage.css';
import SignUpPage from '../../components/SignUpPage';
import { addUser } from '../../actions';
import { mockProfile } from '../../tests/testMocks';

export class ProfilePage extends Component {
    constructor(props){
        super(props)
    }

    componentDidMount(){
        // if(!Object.keys(this.props.currentPage).length){
        //     this.props.history.push('/')
        // }
        this.props.addUser(mockProfile)
    }



    render(){
        const { currentPage, user } = this.props;
        // const header = currentPage.replace('_', ' ').toUpperCase()
        // const lowerCaseBiz = header.toLowerCase();
        if(!Object.keys(currentPage).length && !user){
           return(
               <div>
                <h3>hello</h3>   
               </div>
           )
        } else if(currentPage && user.type === 'food_truck'){
            const { attributes } = this.props.user;
            return(
                <div className='parent-container'>
                    <h2 className='biz-name'>{attributes.name}</h2>
                    <div className='profile-container'>
                        <div className='biz-container'>
                          <h5 className='biz-title'>Food Type:</h5>
                          <h5 className='biz-info'>{attributes.food_type}</h5>

                          <h5 className='biz-title'>Phone Number:</h5>
                          <h5 className='biz-info'>{attributes.phone}</h5>

                          <h5 className='biz-title'>Contact Name:</h5>
                          <h5 className='biz-info'>{attributes.contact_name}</h5>

                          <h5 className='biz-title'>Email:</h5>
                          <h5 className='biz-info'>{attributes.email}</h5>

                          <h5 className='biz-title'>Website:</h5>
                          <a href={attributes.website} target='blank' className='biz-info'>{attributes.website}</a>
                        </div>

                        <div className='upcoming-events-container'>
                          <h3 className='upcoming-events-title'>UPCOMING EVENTS</h3>
                          <h4 className='upcoming-events-date'>12/12 <span className='upcoming-events-status'>Need a Booking</span></h4>
                          <h4 className='upcoming-events-date'>12/13 <span className='upcoming-events-status'>@ Cerebral</span></h4>
                          <h4 className='upcoming-events-date'>12/14 <span className='upcoming-events-status'>@ Diebolt </span></h4>
                          <h4 className='upcoming-events-date'>12/15 <span className='upcoming-events-status'>Need a Booking</span></h4> 
                          <h4 className='upcoming-events-date'>12/16 <span className='upcoming-events-status'>@ Call To Arms</span></h4>
                          <h4 className='upcoming-events-date'>12/17 <span className='upcoming-events-status'>@ Zuni Street Brewery </span></h4>
                          <h4 className='upcoming-events-date'>12/18 <span className='upcoming-events-status'>Need a Booking</span></h4> 
                        </div>
                    </div>
                </div>
            )
        } else if (currentPage && user.type === 'brewery'){
            const { attributes } = this.props.currentPage.data
            return(
                <div>
                    <h3>hi!!!!</h3>
                </div>
            )
        } else {
            return(
                <div>
                </div>
            )
        }
    }
}

export const mapStateToProps = (state) => ({
    currentPage: state.currentPage,
    user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
    addUser: (user) => dispatch(addUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePage)