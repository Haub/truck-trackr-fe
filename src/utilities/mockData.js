export const mockBreweries = [
  
  {
    id: 105,
    name: 'Call to Arms',
    address: '4300 Tennyson Street',
    contact_name: 'Joe Smith',
    phone: '303-333-3333',
    email: 'calltoarms@gmail.com',
    website: 'www.calltoarms.com',
    uuid: 'asdsa1212'
  },

  {
    id: 102,
    name: 'Zuni Street Brewery',
    address: '2900 Zuni Street',
    contact_name: 'George Smith',
    phone: '303-999-9999',
    email: 'zunistreet@gmail.com',
    website: 'www.zunistreet.com',
    uuid: 'sdfssd2322'
  }
];

export const mockBrewery = {
    id: 105,
    name: 'Call to Arms',
    address: '4300 Tennyson Street',
    contact_name: 'Joe Smith',
    phone: '303-333-3333',
    email: 'calltoarms@gmail.com',
    website: 'www.calltoarms.com',
    uuid: 'asdsa1212'
};

export const mockFoodTrucks = [
  {
    id: 33,
    name: 'Little India',
    food_type: 'Indian',
    contact_name: 'Jill Smith',
    phone: '303-222-2222',
    email: 'littleindia@gmail.com',
    website: 'www.littleindia.com',
    uuid: 'sfadasd932'
  },
  
  {
    id: 13,
    name: 'Quiero Arepas',
    food_type: 'Venezuelan',
    contact_name: 'Jack Smith',
    phone: '303-888-9999',
    email: 'quieroarepas@gmail.com',
    website: 'www.quieroarepas.com',
    uuid: 'jdfdjd222'
  }
];

export const mockFoodTruck = {
    id: 13,
    name: 'Quiero Arepas',
    food_type: 'Venezuelan',
    contact_name: 'Jack Smith',
    phone: '303-888-9999',
    email: 'quieroarepas@gmail.com',
    website: 'www.quieroarepas.com',
    uuid: 'jdfdjd222'
};

export const mockProfile = {
  id: 33,
  name: 'Little India',
  food_type: 'Indian',
  contact_name: 'Jill Smith',
  phone: '303-222-2222',
  email: 'littleindia@gmail.com',
  website: 'www.littleindia.com',
  uuid: 'sfadasd932'
};

export const mockEvent = {
  id: 433,
  date: '04/03/2018',
  truck_booked: true,
  brewery_id: 105
};

export const mockUserEvents = [
  {
    id: 433,
    date: '04/03/2018',
    truck_booked: true,
    brewery_id: 105
  },
  {

  }
]
