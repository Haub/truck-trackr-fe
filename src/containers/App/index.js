import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SignUpPage } from '../../components/SignUpPage';
import { BusinessContainer } from '../../components/BusinessContainer';
import './App.css';
import hamburger from '../../assets/hamburger.svg';
import ProfilePage from '../ProfilePage';
import NavBar from '../NavBar';
import truck from '../../assets/food-truck.png';
import barrel from '../../assets/barrel-icon-new.png';
import { withFirebase } from '../../components/Firebase';
import { loadProfile } from '../../actions'

export class App extends Component {
  constructor() {
    super()
    this.state = {
      login: true,
      navOpen: true,
    }
  }

  componentDidMount(){
    if(!this.state.login){
      this.props.history.push('/login')
    }
    if(!this.props.currentPage) {
      this.props.history.push('/')
    }
  }

  handleNavBar = (e) => {
    document.querySelector('.menu-trigger').classList.toggle('active')
    this.setState ({navOpen: !this.state.navOpen})
  }

  render() {
    let navBar;
    const { navOpen } = this.state;
    const { match } = this.props;

    return (
      <div className='main'>
        <header>
          <h1 className='main-title'><span><img src={ truck } className='truck-icon'/></span>TruckTrackr<span><img src={ barrel } className='barrel-icon'/></span></h1>
          <section>
            <a onClick={ this.handleNavBar} className="menu-trigger" id="menu05">
              <span></span>
              <span></span>
              <span></span>
            </a>
          </section>
          
        </header>

        <div className='content-holder'>
        { this.state.navOpen &&
          <NavBar history={this.props.history}/>
        }
        { !this.state.navOpen &&
          <div id="wrapper">
            <span id="openbtn"  onClick={this.handleNavBar}></span>
          </div>
        }
        <Route
          exact path='/login'
          render={ props => (
            <SignUpPage history={this.props.history}/>
          )
           }
        />
        <Route 
          exact path='/'
          component={ ProfilePage }
        />
        <Route 
          exact path='/breweries'
          render={ props => (
            <BusinessContainer 
              data={this.props.breweries} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route 
          exact path='/food_trucks'
          render={ props => (
            <BusinessContainer 
              data={this.props.foodTrucks} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route path='/business/:businessName' component={ ProfilePage } />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  breweries: state.breweries,
  foodTrucks: state.foodTrucks,
  currentPage: state.currentPage
});

export const mapDispatchToProps = (dispatch) => ({
  loadProfile: (profile) => dispatch(loadProfile(profile))
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(withFirebase(App)));