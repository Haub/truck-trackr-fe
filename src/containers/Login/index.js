import React, { Component } from "react";
import { NavLink, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./Login.css";
import { addUser, loadProfile, loadEvents } from "../../actions";
import * as helper from "../../utilities";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      passwordOne: "",
      signUp: false,
      locationType: "",
      businessName: "",
      address: "",
      phoneNumber: "",
      contactName: "",
      foodType: "",
      logo: null
    };
  }

  handleKeyPress = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    // e.preventDefault();
    const { email, passwordOne, signUp, locationType } = this.state;


    if (signUp) {
      this.props.firebase
        .doCreateUserWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          let user = this.cleanUserForSignUp(authUser);
          sessionStorage.setItem("uid", authUser.user.uid);
          let result = await helper.createNewUser(user, locationType);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    } else {
      this.props.firebase
        .doSignInWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          sessionStorage.setItem("uid", authUser.user.uid);
          const user = this.cleanUserForLogin(authUser, locationType);
          const result = await helper.loginUser(user);
          console.log(result);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    }
  };

  cleanUserForSignUp = authUser => {
    const {
      username,
      email,
      passwordOne,
      signUp,
      locationType,
      businessName,
      address,
      phoneNumber,
      contactName,
      foodType
    } = this.state;
    let user;
    if (locationType === "food_trucks") {
      user = {
        name: businessName,
        food_type: foodType,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        website: username,
        uid: authUser.user.uid
      };
    } else {
      user = {
        name: businessName,
        address: address,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        uid: authUser.user.uid
      };
    }
    return user;
  };

  cleanUserForLogin = (authUser, locationType) => {
    let user;
    if (locationType === "food_trucks") {
      user = {
        uid: authUser.user.uid,
        account_type: "food_truck"
      };
    } else {
      user = {
        uid: authUser.user.uid,
        account_type: "brewery"
      };
    }
    return user;
  };

  toggleSignUp = () => {
    let newSignUp = !this.state.signUp;
    this.setState({ signUp: newSignUp });
  };

  fileChangedHandler = event => {
    this.setState({
      logo: event.target.files[0]
    });
  };

  uploadHandler = () => {
    console.log(this.state.logo);

    // const formData = new FormData()
    // formData.append('myFile', this.state.logo, this.state.logo.name)
    // axios.post('my-domain.com/file-upload', formData)
  };

  render() {
    const { signUp, locationType } = this.state;
    let showFoodType;

    if (signUp && locationType === "food_trucks") {
      showFoodType = true;
    }

    return (
      <main className="login">
        <form className="login-form" onSubmit={this.handleSubmit}>
          <input
            className="email-input"
            name="email"
            type="email"
            placeholder="Email"
            onChange={this.handleKeyPress}
          />
          <input
            className="password-input"
            name="passwordOne"
            type="password"
            placeholder="Password"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "business-name-input" : "hidden"}
            name="businessName"
            placeholder="Business Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "address-input" : "hidden"}
            name="address"
            placeholder="Address/City Located"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "phone-number-input" : "hidden"}
            name="phoneNumber"
            placeholder="Phone Number"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "contact-name-input" : "hidden"}
            name="contactName"
            placeholder="Contact Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={showFoodType ? "food-type-input" : "hidden"}
            name="foodType"
            placeholder="Food Type"
            onChange={this.handleKeyPress}
          />
          <p className={signUp ? "business-type" : "hidden"}>
            What Type of Business:
          </p>
          <div className="button-holder">
            <input
              className={"food-truck-radio"}
              type="radio"
              name="locationType"
              value="food_trucks"
              id="food-truck-button"
              onChange={this.handleKeyPress}
            />
            <label className={"food-truck-label"} htmlFor="food-truck-button">
              Food Truck
            </label>
            <input
              className={"brewery-radio"}
              type="radio"
              name="locationType"
              value="breweries"
              id="brewery-button"
              onChange={this.handleKeyPress}
            />
            <label className={"brewery-label"} htmlFor="brewery-button">
              Brewery
            </label>
          </div>
          {/* <input
            className={signUp ? "logo-input" : "hidden"}
            name="logo"
            placeholder="Upload logo"
            onChange={this.handleKeyPress}
            type="file"
          />
          <button
            className={signUp ? "logo-button" : "hidden"}
            onClick={this.uploadHandler}
          >
            Upload!
          </button> */}
          <NavLink
            to="/profile"
            className="signin-button"
            onClick={e => this.handleSubmit(e)}
          >
            {signUp ? "Sign Up" : "Sign In"}
          </NavLink>
        </form>
        <button className="signup-button" onClick={this.toggleSignUp}>
          {signUp
            ? "Actually I already have an account"
            : "First Time Here? Sign Up!"}
        </button>
      </main>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(addUser(user)),
  loadProfile: currentPage => dispatch(loadProfile(currentPage)),
  loadEvents: events => dispatch(loadEvents(events)) 
});

const { object, func } = PropTypes;

Login.propTypes = {
  user: object,
  addUser: func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Login));
