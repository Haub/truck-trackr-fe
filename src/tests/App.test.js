import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from '../containers/App/';
import { App } from '../containers/App/';
import { mockProfile } from '../utilities/mockData.js';


describe ('APP', () => {
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state login and navOpen set to true ', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.state().login).toEqual(true);
    expect(wrapper.state().navOpen).toEqual(true);
  });

  describe('mapStateToProps', () => {
    it('should have access to breweries, foodTrucks, and currentPage', () => {
      const mockStore = {
        breweries: [],
        foodTrucks: [],  
        currentPage: {}
      }
      const expected = {...mockStore};
      const result = mapStateToProps(mockStore);
      expect(result).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when loadProfile is invoked', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadProfile(mockProfile);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });

});

