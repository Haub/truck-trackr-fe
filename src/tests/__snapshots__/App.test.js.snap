// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the snapshot 1`] = `
<div
  className="main"
>
  <header>
    <h1
      className="main-title"
    >
      <span>
        <img
          className="truck-icon"
          src="food-truck.png"
        />
      </span>
      TruckTrackr
      <span>
        <img
          className="barrel-icon"
          src="barrel-icon-new.png"
        />
      </span>
    </h1>
    <section>
      <a
        className="menu-trigger"
        id="menu05"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </a>
    </section>
  </header>
  <div
    className="content-holder"
  >
    <Connect(NavBar) />
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/breweries"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/food_trucks"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/business/:businessName"
    />
  </div>
</div>
`;
